File: sources/clojure-api/src/clojure_api/features_service.clj
(ns clojure-api.features-service
  (:require [clojure-api.mongo-service :as mongo]))

(defn get-features []
  (let [result (mongo/find-documents "featureFlags" {})]
    (if (not (empty? result))
      (reduce (fn [acc {:keys [name value]}] (assoc acc name value)) {} result)
      [])))

File: sources/express-api/src/services/featuresService.js
const { findDocuments } = require('../db/mongo');

async function getFeatures() {
  const docs = await findDocuments('featureFlags', {});
  if (!docs.length) {
    return { status: 404, message: 'No features found' };
  }
  // Convert [{name, value}] to { name: value, ... }
  const features = docs.reduce((acc, { name, value }) => {
    acc[name] = value;
    return acc;
  }, {});
  return features;
}

module.exports = { getFeatures };
